sparsehash/internal/densehashtable.h

```diff
@@ -779,6 +779,12 @@ class dense_hashtable {
     return *this;
   }
 
+  dense_hashtable& operator=(dense_hashtable&& ht) {
+    if (&ht == this) return *this;  // don't copy onto ourselves
```

Comment:
I don't think this could be ever true. `ht` is a rvalue - so a temporary - how could it be this?

The only case where this can happen is with a code like `h = std::move(h)`, which is wrong de facto. As a trade off, you could replace that by an assert?